src/components/TextInput/TextInput.tsx

```diff
@@ -2,12 +2,25 @@ import React, { ChangeEvent, FunctionComponent } from 'react';
 
 import { Icon } from '../Icon/Icon';
 
+type Type =
+	| 'button'
```

Comment:
Dit zijn alle types voor een `<input>`, maar deze component is de TextInput, zou je dit niet limiteren tot `text | password`?